pyuvsim/spherical_coordinates_basis_transformation.py

```diff
@@ -0,0 +1,295 @@
+# -*- mode: python; coding: utf-8 -*
+# Copyright (c) 2018 Radio Astronomy Software Group
+# Licensed under the 3-clause BSD License
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+
+def r_hat(theta, phi):
+    """
```

Comment:
This is equivalent to the `ang2vec` function in `healpy`, but doesn't appear to be available yet in astropy-healpix. In astropy, there's `astropy.coordinates.UnitSphericalRepresentation`.

Not strictly important, but I figure it's worth mentioning.

Reply:
ok, I haven't looked at `astropy.coordinates.UnitSphericalRepresentation`. Are you suggesting we use that instead of this?

Reply:
Hmm, that does look like it might do what we want. I'll test with it a little. @jaguirre you might like to know about this too.

Reply:
If this works, I don't see the harm in using it. It might also be faster than calling an external library, depending on how it's used. I ran some short tests and it looks like the astropy method is somewhat slower:
```
Astropy : 3.39 ms ± 0 ns per loop (mean ± std. dev. of 1 run, 1 loop each)
r_hat : 232 µs ± 0 ns per loop (mean ± std. dev. of 1 run, 1 loop each)
```

Reply:
I looked at it more, it doesn't quite do what we want anyway. It only gives the unit vectors along lat & lon, not along r, and lat is 90 degrees off of theta, which we use throughout this code. So given that it's fast, I think we should stick with what we have.